pkg/cmd/pulumi/package_add.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	"github.com/spf13/cobra"
+)
+
+// Constructs the `pulumi package add` command.
+func newPackageAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add <provider> [provider parameters]",
```

Comment:
Are `[provider parameters]` space separated strings, or are they cli args like `--something` ? Would we need  `--` to disambiguate? `add <provider> -- [provider parameters]`

Reply:
Space-separated but I guess the provider can interpret them arbitrarily. So if you want to pass flags then you'd have to use `--`. Thoughts on how we message this/do we have examples of communicating this elsewhere do you know?

Reply:
> Thoughts on how we message this/do we have examples of communicating this elsewhere do you know?

Not that I know, but I know very little about pulumi 😄 Maybe the instructions should tell users to always use `--` , but that might also be slightly confusing for users and some might not realise it's a standalone `--` with a space after it.

Reply:
Fine as is for now I think!